server/api/src/main/java/org/hawkular/apm/server/api/utils/ProducerInfoUtil.java

```diff
@@ -196,6 +197,10 @@ public static ProducerInfo getProducerInfo(String tenantId, Span serverSpan, Spa
                 pi.getProperties().addAll(clientSpan.binaryAnnotationMapping().getProperties());
                 pi.setHostAddress(clientSpan.ipv4());
 
+                if (clientSpan.service() != null) {
+                    pi.getProperties().add(new Property(Constants.PROP_SERVICE_NAME, serverSpan.service()));
```

Comment:
should be 'clientSpan' - sorry was not clear that both lines needed to be changed